packages/router-generator/src/generator.ts

```diff
@@ -55,6 +55,10 @@ async function getRouteNodes(config: Config) {
         return d.name.startsWith(routeFilePrefix)
       }
 
+      if (routeFileIgnoreType) {
+        return !d.name.match(new RegExp(routeFileIgnoreType, 'g'))
```

Comment:
create the regex only once

Reply:
https://github.com/TanStack/router/pull/1310/commits/c4e21de9620695d1f3329f773de3be499de36b5c
thx